ic-log/src/canister/state.rs

```diff
@@ -3,61 +3,36 @@ use std::cell::RefCell;
 
 use candid::{Decode, Encode, Principal};
 use ic_stable_structures::stable_structures::DefaultMemoryImpl;
-use ic_stable_structures::{Bound, CellStructure, IcMemoryManager, MemoryId, StableCell, Storable};
+use ic_stable_structures::{Bound, CellStructure, StableCell, Storable, VirtualMemory};
 use ic_storage::IcStorage;
 
 use crate::did::{LogCanisterError, LogCanisterSettings, LoggerAcl, LoggerPermission, Pagination};
 use crate::writer::{InMemoryWriter, Logs};
 use crate::{take_memory_records, LogSettingsV2, LoggerConfig};
 
 thread_local! {
-    static MEMORY_MANAGER: IcMemoryManager<DefaultMemoryImpl> = IcMemoryManager::init(DefaultMemoryImpl::default());
     static LOGGER_CONFIG: RefCell<Option<LoggerConfig>> = const { RefCell::new(None) };
```

Comment:
is it not required to store the LoggerConfig too? If not, why do we need this thread local variable?

Reply:
`LoggerConfig` contains a reference to the filter that the `log` operations use, so that we can update the logger filter while the canister is running. We cannot (and don't need to) store it in the stable memory. We store logger filter string as a part of the `settings` field, and use it to reset `LoggerConfig` when a canister being upgraded.